### DISCUSSION TITLE
Some doubts regarding the Gitlab integration

### DISCUSSION THREAD

**Discussion by haaris94** [ueman/feedback#204] [Q&A] [answered]
Hi! I am new to Gitlab, so I am struggling a bit here. So my questions are as follows

1. For project id, I got it from the project settings
2. For api-token, I created a Personal access token from **Edit Profile** -> **Access tokens** -> Create a new personal token with read_api and write_repository access

And I put the code as follows, I don't give any value to Gitlab url
```
BetterFeedback.of(context).showAndUploadToGitLab(
  projectId: 'project-id',
  apiToken: 'api-token',
);
```
Once I click submit, it stays like that for a second and returns to the normal app screen. But I don't see any issues created in my project

**Comment by ueman** [comment, marked as answer]
If you're using gitlab saas, you don't need to specify any url. What did you use as `projectId`, also do you get any error logged to console?

**Reply by haaris94** [reply]
Yes I am using gitlab saas

What did you use as `projectId`?
It was a number with 8 char. Something along the lines of "69532178". I got it from my Project -> Settings -> General -> Project ID

do you get any errors logged to the console?
Nothing.

**Reply by ueman** [reply]
Those settings sound correct to me. Which means you have to debug it yourself. 
Especially look into the response of this [request](https://github.com/ueman/feedback/blob/6263bbd887f782cb48e52ca3a655c29717d904e6/feedback_gitlab/lib/src/feedback_gitlab.dart#L62) and also at this [response](https://github.com/ueman/feedback/blob/6263bbd887f782cb48e52ca3a655c29717d904e6/feedback_gitlab/lib/src/feedback_gitlab.dart#L86). Those should contain information about whether the request was successful. You can do that by browsing the code and setting a breakpoint.

**Reply by haaris94** [reply]
Ok I set the breaking point in the `uploadToGitLab` method after `uploadResponse` json decode and the following is the response

UploadResponse: Status code = 403 &  responsePhrase = Forbidden

UploadResponseMap: "error" -> "insufficient_scope", "error_description" -> "The request requires higher privileges than provided by the access token.", "scope" -> "api read_api".

I'll create a new `api-token` and try again. Thank you. But, why is this not throwing?

**Reply by ueman** [reply]
Because it's not programmed that way, but it would be a good addition. I would gladly accept a PR which adds that.

**Reply by haaris94** [reply]
Success!! So I created a new `api-token` and gave it full api access and it works flawlessly. The feedback is registered in the Issues section.

The only problem right now is that once I click submit button, the screen freezes for 1-2 secs and I can't show some kind of loading indicator. Is there any method that I can reliably `await `to show loading?

**Reply by haaris94** [reply]
> Because it's not programmed that way, but it would be a good addition. I would gladly accept a PR which adds that.

I am sorry. I am new to this whole Git stuff, so it is beyond my capabilities. In fact, I don't even know what PR means.
